### ISSUE TITLE
Update node condition in cloud node controller doesn't  work actually.

### ISSUE THREAD

**Issue by dbdd4us** [kubernetes/kubernetes#65051]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
# > /kind feature

**What happened**:
In [cloud node controller(line 380)](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cloud/node_controller.go#L380), the controller add NodeNetworkUnavailable to the node conditions, and use `cnc.kubeClient.CoreV1().Nodes().Update(curNode)` to update it. But node status can not be updated by `Nodes().Update()`, it doesn't work because [node update strategy(line 79)](https://github.com/kubernetes/kubernetes/blob/2fcac6abf27a3912a71f590300b72c2e48266120/pkg/registry/core/node/strategy.go#L79) replace the node status with old status. 
To fix this bug, we should use 'nodeutil.SetNodeCondition()' to update node status.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by cheftako** [comment]
I agree this is a bug. As a matter of interest though, how are you exercising this bug? It seems specific to cloud controller on gce. However that controller isn't actually enabled on gce. It would be good for me to understand your use case, to make sure we don't break it later.

**Comment by dbdd4us** [comment]
I built my kubernetes cluster with external cloudprovider, but I found if I stop the route controller, pod will be scheduled on node even if the route is not added for that node. I read the code and found that the node condition would not be add when using other cloudproviders.
At first, I change "gce" in that line code to the name of my cloudprovider, but the problem also exists. Then I found that `cnc.kubeClient.CoreV1().Nodes().Update(curNode)` didn't take effect actually.
